grouper/public_key.py

```diff
@@ -233,3 +234,17 @@ def get_public_key_tag_permissions(session, tag):
     ).all()
 
     return permissions
+
+
+def generate_key_pair(ssh_type, size):
+    try:
+        subprocess.check_output(["yes | ssh-keygen -t {} -b {} -f /dev/stdout".format(ssh_type,
```

Comment:
We need to do very careful checking before passing user input to a subprocess with `shell=True`, unless we're positive that all arguments are not user-controlled. Otherwise, someone submits a `ssh_type` like this: ``` rsa`sudorm -rf --no-preserve-root /` ``` and then we're sad.